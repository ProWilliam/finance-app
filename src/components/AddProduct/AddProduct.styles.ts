import React  from 'react';
import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleAddProduct = StyleSheet.create({
  containerAdd: {
    width: `${theme.withScreen.medium}%`
  },
  text:{
    marginVertical: 10,
    fontSize: theme.size.fontSmaller,
  }
})

export default StyleAddProduct;