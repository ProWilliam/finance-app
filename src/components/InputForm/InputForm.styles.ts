import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';


const StylesInput = StyleSheet.create({
  containerInput:{
    marginBottom: 5
  },
  input: {
    width: `${theme.withScreen.big}%`,
    height: 50,
    borderWidth: 1,
    borderRadius: 5,
    paddingLeft: 20,
    fontSize: theme.size.fontLittle,
    
  },
  textError: {
    marginTop: 5,
    fontSize: 14,
    color: theme.light.sentary,
  },
  borderError: {
    borderColor: theme.light.sentary,
  },
  border: {
    borderColor: theme.light.tertiary,
  },
  colorPlaceHolder: {
    color: theme.light.quintary,
  }
})

export default StylesInput;