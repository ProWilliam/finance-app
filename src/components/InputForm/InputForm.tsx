import React, { useState } from 'react';
import { TextInput, View, Text, KeyboardTypeOptions } from 'react-native';

// Components
import InputDate from '../InputDate/InputDate';

// Styles and Types
import styles from './InputForm.styles';
import InputFormProps from '../../types/components/InputForm.types';


const InputForm: React.FC<InputFormProps> = ({placeHolder, keySelect, errorMessage, onChange, type, value }) => {

  const [error, setError] = useState<string | null>(errorMessage || null);
  const [text, setText] = useState<string>('');

  const validatorText = () => {
    if(text.trim() === ''){
      setError('Este campo es obligatorio!');
    } else {
      setError(null);
    }
  }

  const handleChangeText = (value: string) => {

    setText(value);
    onChange(keySelect, value);

    if (value.trim() !== '') {
      setError(null);
    }else {
      setError('Este campo es obligatorio!');
    }
  }

  // Change of styles for action
  const inputStyles = [
    styles.input,
    error !== null
    ? styles.borderError 
    : styles.border,
  ]

  return (
    <View style={styles.containerInput}>
      {
        type ==='date'
        ? <InputDate 
            title={placeHolder}
            keyValue={keySelect}
            onChangeForm={onChange}
            value={value}
          />
        :
        <TextInput 
          style={inputStyles} 
          placeholder={placeHolder}
          placeholderTextColor={styles.colorPlaceHolder.color}
          onBlur={validatorText}
          onChangeText={(text) => handleChangeText(text)}
          value={value}
          blurOnSubmit 
          keyboardType={type as KeyboardTypeOptions}
          testID='error-message'
        /> 

      }      

      { error && <Text testID='error-message' style={styles.textError}>{error}</Text> }
    </View>
  )
}

export default InputForm;