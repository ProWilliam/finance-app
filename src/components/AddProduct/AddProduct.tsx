import React from 'react';
import { View, Text } from 'react-native';

// Components
import InputForm from '../InputForm/InputForm';

// Styles and Types
import styles from './AddProduct.styles';
import { AddProductProps } from '../../types/components/AddProduct.types';


const AddProduct: React.FC<AddProductProps> = ({title, ...props}) => {
  return(
    <View style={styles.containerAdd}>
      <Text style={styles.text}>
        {title}
      </Text>
      <InputForm {...props} />
    </View>
  )
}

export default AddProduct;