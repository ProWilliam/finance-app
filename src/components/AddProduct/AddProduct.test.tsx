import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

// Components
import AddProduct from './AddProduct';
import InputForm from '../InputForm/InputForm';

// Types
import { AddProductProps } from '../../types/components/AddProduct.types';

describe('<AddProduct />', () => {

  let props: AddProductProps;

  it('Render the title correctly with props ID', () => {
    props = {
      value: '1',
      title: 'ID', 
      placeHolder: 'Numero de ID',
      type: 'numeric',  
      keySelect: 'id',
      onChange: () => {},
    }
    const tree = renderer.create(<AddProduct {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('Renders the title correctly', () => {
    props = {
      value: 'New Product',
      title: 'Nombre', 
      placeHolder: 'Tarjeta CrÃ©dito',
      type: 'default',  
      keySelect: 'name',
      onChange: () => {},
    } 

    const component = renderer.create(<AddProduct {...props} />);
    const instance = component.root;
    
    const text = instance.findByType(Text);
    expect(text.props.children).toBe('Nombre');

  });

  it('Renders the InputForm component', ()=> {
    const component = renderer.create(<AddProduct {...props} />);
    const instance = component.root;
    const inputForm = instance.findByType(InputForm);

    // Verify that the `InputForm` component exists.
    expect(inputForm).toBeTruthy();
  });
  
});