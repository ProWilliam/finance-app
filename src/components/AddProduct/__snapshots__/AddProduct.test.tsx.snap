// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddProduct /> Render the title correctly with props ID 1`] = `
<View
  style={
    {
      "width": "80%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 14,
        "marginVertical": 10,
      }
    }
  >
    ID
  </Text>
  <View
    style={
      {
        "marginBottom": 5,
      }
    }
  >
    <TextInput
      blurOnSubmit={true}
      keyboardType="numeric"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Numero de ID"
      placeholderTextColor="#6a6a6a"
      style={
        [
          {
            "borderRadius": 5,
            "borderWidth": 1,
            "fontSize": 24,
            "height": 50,
            "paddingLeft": 20,
            "width": "100%",
          },
          {
            "borderColor": "#eaeaea",
          },
        ]
      }
      testID="error-message"
      value="1"
    />
  </View>
</View>
`;
